src/Data/StrMap.purs

```diff
@@ -235,6 +236,9 @@ fromFoldableWith f l = pureST (do
   for_ l (\(Tuple k v) -> runFn4 _lookupST v (f v) k s >>= SM.poke s k)
   pure s)
 
+-- | Create a map from a homogeneous record (all attributes have the same type).
+foreign import fromRecord :: forall r t. Homogeneous r t => Record r -> StrMap t
```

Comment:
I don't think it's a good idea for foreign functions to take constraints. It's better to declare an "unsafe" `impl` function for internal use, and add a constraint to the PS wrapper. Honestly, I wish PS altogether disallowed constraint syntax in foreign imports.

Reply:
Thanks. Will fix.